config-schema.json

```diff
@@ -121,23 +125,25 @@
       "description": "Multiple named projects",
       "properties": {
         "projects": {
-          "additionalProperties": {
-            "anyOf": [
-              {
-                "$ref": "#/definitions/IGraphQLProject"
-              },
-              {
-                "$ref": "#/definitions/IGraphQLProjectLegacy"
-              }
-            ]
-          },
-          "title": "projects",
-          "type": "object"
+          "$ref": "#/definitions/Record<string,IGraphQLProject|IGraphQLProjectLegacy>",
+          "title": "projects"
         }
       },
       "required": ["projects"],
       "title": "IGraphQLProjects",
       "type": "object"
+    },
+    "Record<string,IGraphQLProject|IGraphQLProjectLegacy>": {
+      "title": "Record<string,IGraphQLProject|IGraphQLProjectLegacy>",
+      "type": "object"
```

Comment:
I think this should have a dynamic property like;
```
            "patternProperties": {
                ".*": {
                    "anyOf": [
                        { "$ref": "IGraphQLProject" },
                        { "$ref": "IGraphQLProject" }
                     ]
              }
 ```
 See https://json-schema.org/understanding-json-schema/reference/object.html#pattern-properties

Reply:
Maybe somethign like this;
https://github.com/kamilkisiela/graphql-config/pull/1011/files#diff-df4af8dbe3d6fbc388b8276ca4494407f03a6723884ee3d1f7e03f754aa241beL124

Reply:
this file was generated by `typescript-json-schema` package

Reply:
@ardatan I think it's not supported by `typescript-json-schema`
Tried to generate the following types
<img width="692" alt="image" src="https://user-images.githubusercontent.com/7361780/161113268-c38bd68d-715b-41bc-aa96-f203edabe601.png">
Received the following error
```
/Users/dimitri/Desktop/GUILD/graphql-config/node_modules/typescript-json-schema/dist/typescript-json-schema.js:787
                        throw new Error("Not supported: IndexSignatureDeclaration with index symbol other than a number or a string");
                              ^

Error: Not supported: IndexSignatureDeclaration with index symbol other than a number or a string

```

Reply:
But we will lose the typesafety we had before in this case. 
https://github.com/kamilkisiela/graphql-config/pull/1011/files#diff-df4af8dbe3d6fbc388b8276ca4494407f03a6723884ee3d1f7e03f754aa241beL124

Reply:
@ardatan seems like `typescript-json-schema` doesn't play well with the `Record` type, I made rollback to use interfaces with index signature https://github.com/YousefED/typescript-json-schema/issues/337

Could you re-review it? 🙂